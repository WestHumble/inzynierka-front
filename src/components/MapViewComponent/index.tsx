export { default as MapViewComponent } from "./MapViewComponent";
