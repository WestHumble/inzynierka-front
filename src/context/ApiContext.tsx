import React, {createContext, useState} from 'react';
import axios from 'axios';
import {useNavigation} from "@react-navigation/native";
export const ApiContext = createContext(null);

const API_ADDRESS = process.env.EXPO_PUBLIC_API_URL;
export const ApiProvider = ({children}) => {
    const [userToken, setUserToken] = useState<string>();
    const [userRefreshToken, setUserRefreshToken] = useState<string>();
    let axiosAuthInstance, axiosNoAuthInstance

    const deleteInstances = ()=>{
        axiosAuthInstance = null
        axiosNoAuthInstance = null
    }
    async function getInstance(noAuth, successCallback, errorCallback) {
        if (noAuth) {
            if (!axiosNoAuthInstance) {
                axiosNoAuthInstance = axios.create({
                    baseURL: `${API_ADDRESS}`,
                })
            }
            return axiosNoAuthInstance
        }
        if (!axiosAuthInstance) {
            axiosAuthInstance = axios.create({
                baseURL: `${API_ADDRESS}`,
                headers: {
                    Authorization: `Bearer ${userToken}`,
                }
            })
            axiosAuthInstance.interceptors.response.use(
                response => {
                    return response;
                },
                error => {
                    if (error.request && error.request.status !== 401) {
                        throw error;
                    }
                    let failedRequest = error.config;
                    axios.create({
                        baseURL: `${API_ADDRESS}`,
                    }).post('api/token/refresh', {
                        refresh_token: userRefreshToken
                    }).then((res)=>{
                        setUserToken(res.data.token)
                        setUserRefreshToken(res.data.refresh_token)
                        axiosAuthInstance.defaults.headers['Authorization'] = 'Bearer ' + res.data.token;
                        failedRequest.headers['Authorization'] = 'Bearer ' + res.data.token;

                        return axios.request(failedRequest)
                            .then(res => {
                                if (!res || null === successCallback){
                                    return
                                }
                                successCallback(res)
                            })
                            .catch(res => {
                                if (!res || null === errorCallback){
                                    return
                                }
                                errorCallback(res)
                            })
                    }).catch((res)=>{
                        console.error("Unable to refresh token")
                        if (!res || null === errorCallback){
                            return
                        }
                        errorCallback(res)
                    })
                },
            );
        }
        return axiosAuthInstance
    }
    const post = async (endpoint, body? = null, success = null, error = null, noAuth = false)=> {
        if (!noAuth && !userToken){
            return
        }
        (await getInstance(noAuth, success, error)).post(`/api/${endpoint}`, body).then(res => {
            if (!res || null === success){
                return
            }
            success(res)
        }).catch(res => {
            if (!res || null === error){
                return
            }
            error(res)
        })
    }
    const get = async (endpoint, params? = null, success = null, error = null, noAuth = false)=> {
        if (!noAuth && !userToken){
            return
        }
        (await getInstance(noAuth, success, error)).get(`/api/${endpoint}`, {
            params: params
        }).then(res => {
            if (!res || null === success){
                return
            }
            success(res)
        }).catch(res => {
            if (!res || null === error){
                return
            }
            error(res)
        })
    }
    return (
        <ApiContext.Provider value={{post, get, deleteInstances, userToken, setUserToken, setUserRefreshToken}}>
            {children}
        </ApiContext.Provider>
    );
}