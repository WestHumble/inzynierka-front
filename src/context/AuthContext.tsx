import AsyncStorage from '@react-native-async-storage/async-storage';
import axios from 'axios';
import React, {createContext, useEffect, useState} from 'react';
import { login as loginPost } from "../components/Api/login";


export const AuthContext = createContext(null);

export const AuthProvider = ({children}) => {
    const [isLoading, setIsLoading] = useState(false);
    const [userToken, setUserToken] = useState<string>();

    const login = (email: string, password: string) => {
        setIsLoading(true)
        loginPost(email, password).then(res => {
            setUserToken(res.data.token)
            AsyncStorage.setItem('userToken', res.data.token) 
            setIsLoading(false)
        }).catch(res => {
            console.error(res) 
            setIsLoading(false)
        })
        // axios.post(`http://localhost:81/api/login`, {
        //     email,
        //     password
        // }).then(res => {
        //     setUserToken(res.data.token)
        //     AsyncStorage.setItem('userToken', res.data.token) 
        //     setIsLoading(false)
        // }).catch(res => {
        //     console.error(res) 
        //     setIsLoading(false)
        // })
    }

    const logout = () => {
        setIsLoading(true)
        setUserToken(null)
        AsyncStorage.removeItem('userToken')
        setIsLoading(false)
    }

    const isLoggedIn = async() => {
        try {
            setIsLoading(true)
            let userToken = await AsyncStorage.getItem('userToken')
            setUserToken(userToken == 'null' ? null : userToken)
            setIsLoading(false)
        } catch (e){
            console.log(`isLoggedIn error ${e}`)
        }

    }

    useEffect(() => {
        isLoggedIn()
    }, [])

    return (
        <AuthContext.Provider value={{login, logout, isLoading, userToken}}>
            {children}
        </AuthContext.Provider>
    );
}