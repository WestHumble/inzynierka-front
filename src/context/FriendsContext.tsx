import React, {createContext, useContext, useEffect, useRef, useState} from 'react';
import {ApiContext} from "./ApiContext";
import {AppState} from "react-native";
import {Friend} from "../types/friend";
import * as Notifications from "expo-notifications";
import {FriendInvite} from "../types/friendInvite";
import {NavigationContext} from "./NavigationContext";

export const FriendsContext = createContext(null);

export const FriendsProvider = ({children}) => {
    const [friends, setFriends] = useState<Friend[]>([]);
    const [pendingInvites, setPendingInvites] = useState<FriendInvite[]>([]);
    const [unreadFriendInvitesCounter, setUnreadFriendInvitesCounter] = useState<number>(0);

    const { get, userToken } = useContext(ApiContext);
    const { navigationRef } = useContext(NavigationContext);
    const getFriendList = () => {
        get('user/get-friends', null, (res) => {
            setFriends(res.data)
        })
    };
    const getPendingFriendInvites = () => {
        get('friend-invite/invites', null, (res) => {
            setPendingInvites(res.data)
        })
    };
    const notificationListener = useRef();
    const responseListener = useRef();

    useEffect(() => {
        notificationListener.current = Notifications.addNotificationReceivedListener(notification => {
            if ('new_friend' === notification.request.trigger.channelId) {
                setTimeout(getFriendList, 1000)
            }
            if ('new_friend_invite' === notification.request.trigger.channelId) {
                setTimeout(getPendingFriendInvites, 1000)
            }
        });

        responseListener.current = Notifications.addNotificationResponseReceivedListener(response => {
            if ('new_friend_invite' === response.notification.request.trigger.channelId) {
                navigationRef.current?.navigate('FriendInvitesScreen');
            }
        });

        return () => {
            Notifications.removeNotificationSubscription(notificationListener.current);
            Notifications.removeNotificationSubscription(responseListener.current);
        };
    }, []);

    useEffect(() => {
        if (userToken) {
            getFriendList()
            getPendingFriendInvites()
        } else {
            setFriends([])
            setPendingInvites([])
        }
    }, [userToken]);

    useEffect(() => {
        setUnreadFriendInvitesCounter(pendingInvites.filter(c=> c.status === 'new').length)
    }, [pendingInvites]);

    return (
        <FriendsContext.Provider value={{
            friends,
            getFriendList,
            pendingInvites,
            getPendingFriendInvites,
            unreadFriendInvitesCounter,
        }}>
            {children}
        </FriendsContext.Provider>
    );
}