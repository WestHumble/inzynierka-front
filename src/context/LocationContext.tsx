import React, {createContext, useContext, useEffect, useState} from 'react';
import * as Location from "expo-location";
import {ApiContext} from "./ApiContext";

export const LocationContext = createContext(null);

export const LocationProvider = ({children}) => {
    const [shareLocation, setShareLocation] = useState(false);
    const [userLocation, setUserLocation] = useState(null);
    const {post, userToken} = useContext(ApiContext)
    let lastUpdateDate = null;

    useEffect(() => {
        post(shareLocation ? 'user/location/init' : 'user/location/stop');
    }, [shareLocation]);

    useEffect(() => {
        if (!userToken) {
            return
        }
        getLocation(shareLocation);
        const interval = setInterval(() => getLocation(shareLocation), parseInt(process.env.EXPO_PUBLIC_LOCALIZATION_REFRESH_TIME));
        return () => clearInterval(interval);
    }, [userToken, shareLocation]);

    const getLocation = async (shareLocation) => {
        if (!userToken) {
            return
        }
        let { status } = await Location.requestForegroundPermissionsAsync();
        if (status !== 'granted') {
            console.log('Permission to access location was denied');
            return;
        }

        let location = await Location.getCurrentPositionAsync({});
        if (!location) {
            return;
        }
        setUserLocation(location);
        if (shareLocation && (null === lastUpdateDate || ((new Date()) - lastUpdateDate) >= parseInt(process.env.EXPO_PUBLIC_LOCALIZATION_UPDATE_TIME))) {
            lastUpdateDate = new Date()
            post('user/location/update', {
                longitude: location.coords.longitude,
                latitude: location.coords.latitude
            }, null, (error)=>{
                if (error.response.status == 404) {
                    setShareLocation(false)
                }
            })
        }
    }


    return (
        <LocationContext.Provider value={{userLocation, shareLocation, setShareLocation}}>
            {children}
        </LocationContext.Provider>
    );
}