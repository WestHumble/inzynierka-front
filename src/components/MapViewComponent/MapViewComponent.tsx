import { StyleSheet } from "react-native";
import React, { useRef, useEffect, useState, useContext } from "react";
import MapView, { Marker, PROVIDER_GOOGLE } from "react-native-maps";
import CustomButton from "../../components/CustomButton";
import {LatLngData, MarkerData} from "../../types/marker";
import { LocationContext } from "../../context/LocationContext";
import {ApiContext} from "../../context/ApiContext";

const MapViewComponent = () => {
  const { userLocation, shareLocation, setShareLocation } =
    useContext(LocationContext);
    const { get, post, userToken } = useContext(ApiContext);
  const mapViewRef = useRef<MapView>(null);
  const [region, setRegion] = useState({
    latitude: userLocation?.coords.latitude ?? 52.4064,
    longitude: userLocation?.coords.longitude ?? 16.9252,
    latitudeDelta: 0.14,
    longitudeDelta: 0.16,
  });

  const [markers, setMarkers] = useState<MarkerData[]>([]);
  const [friendsMarkers, setFriendsMarkers] = useState<MarkerData[]>([]);

  const onRegionChange = (newRegion) => {
    setRegion(newRegion);
  };
    const onShareLocationToggle = () => {
        setShareLocation(!shareLocation);
    };
    const onMercureTest = () => {
        console.log("test3")
    };
  const onLoadCloseEvents = () => {
      post('event/close-list', {
          latitude: region.latitude,
          longitude: region.longitude,
          distanceInMeters: 1000
      }, (res) => setMarkers(res.data))
  };

  useEffect(() => {
    if (mapViewRef.current) {
      mapViewRef.current.animateToRegion(region, 1000);
    }
  }, []);

  const updateFriendsMarkers = () => {
      get('user/location/get-friends', null, (res) => {
          let latlngData : LatLngData
          let newMarkers : MarkerData[] = []
          for(var k in res.data) {
              latlngData = res.data[k]
              newMarkers.push({ latlng: latlngData, name: k, description: "friend" })
          }
          setFriendsMarkers(newMarkers)
      })
  }

    useEffect(() => {
        if (!userToken) {
            return
        }
        updateFriendsMarkers();
        const interval = setInterval(() => updateFriendsMarkers(), parseInt(process.env.EXPO_PUBLIC_LOCALIZATION_UPDATE_TIME));
        return () => clearInterval(interval);
    }, [userToken]);

  return (
    <>
      <MapView
        style={styles.map}
        provider={PROVIDER_GOOGLE}
        initialRegion={region}
        onRegionChange={onRegionChange}
        showsUserLocation={true}
        userInterfaceStyle="dark"
        followsUserLocation={true}
        ref={mapViewRef}
      >
        {markers.map((marker, index) => (
          <Marker
            key={index}
            coordinate={marker.latlng}
            title={marker.name}
            description={marker.description}
          />
        ))}
      {friendsMarkers.map((marker, index) => (
          <Marker
              key={index}
              coordinate={marker.latlng}
              title={marker.name}
              description={marker.description}
          />
      ))}
      </MapView>
      <CustomButton
        additionalStyles={{
          position: "absolute",
          top: "70%",
          left: "35%",
          width: "20%",
          marginHorizontal: "5%",
        }}
        text={shareLocation ? "Sharing on" : "Sharing off"}
        onPress={onShareLocationToggle}
        type="PRIMARY"
        bgColor={undefined}
        fgColor={undefined}
      />

        <CustomButton
            additionalStyles={{
                position: "absolute",
                top: "70%",
                left: "10%",
                width: "20%",
                marginHorizontal: "5%",
            }}
            text={"Mercure test"}
            onPress={onMercureTest}
            type="PRIMARY"
            bgColor={undefined}
            fgColor={undefined}
        />
      {/*<CustomButton*/}
      {/*  additionalStyles={{*/}
      {/*    position: "absolute",*/}
      {/*    top: "89%",*/}
      {/*    left: "70%",*/}
      {/*    width: "20%",*/}
      {/*    marginHorizontal: "5%",*/}
      {/*  }}*/}
      {/*  text="Load close events"*/}
      {/*  onPress={onLoadCloseEvents}*/}
      {/*  type="PRIMARY"*/}
      {/*  bgColor={undefined}*/}
      {/*  fgColor={undefined}*/}
      {/*/>*/}
    </>
  );
};

const styles = StyleSheet.create({
  map: {
    width: "100%",
    height: "100%",
    position: "absolute",
    zIndex: -1,
    elevation: 1,
  },
});

export default MapViewComponent;
