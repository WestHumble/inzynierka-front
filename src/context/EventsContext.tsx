import React, {createContext, useContext, useEffect, useState} from 'react';
import * as Location from "expo-location";
import {ApiContext} from "./ApiContext";
import {LocationContext} from "./LocationContext";
import {MarkerData} from "../types/marker";
import {Event} from "../types/event";

export const EventsContext = createContext(null);

export const EventsProvider = ({children}) => {
    const [events, setEvents] = useState<Event[]>([]);

    const { userLocation, shareLocation, setShareLocation } =
        useContext(LocationContext);
    const { post, userToken } = useContext(ApiContext);
    const loadCloseEvents = (region) => {
        post('event/close-list', {
            latitude: region?.latitude??userLocation?.coords.latitude ?? 52.4064,
            longitude: region?.longitude??userLocation?.coords.longitude ?? 16.9252,
            distanceInMeters: 1000
        }, (res) => setEvents(res.data))
    };

    useEffect(() => {
        if (userToken) {
            loadCloseEvents()
        } else {
            setEvents([])
        }
    }, [userToken]);
    return (
        <EventsContext.Provider value={{events, loadCloseEvents}}>
            {children}
        </EventsContext.Provider>
    );
}