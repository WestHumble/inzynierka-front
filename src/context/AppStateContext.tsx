import React, {createContext, useContext, useEffect, useRef, useState} from 'react';
import axios from 'axios';
import {useNavigation} from "@react-navigation/native";
import {AppState} from "react-native";
import {EventsContext} from "./EventsContext";
import {FriendsContext} from "./FriendsContext";
export const AppContext = createContext(null);

export const AppProvider = ({children}) => {
    const {getPendingEventInvites} = useContext(EventsContext);
    const {getFriendList, getPendingFriendInvites} = useContext(FriendsContext);
    const appState = useRef(AppState.currentState);

    useEffect(() => {
        const subscription = AppState.addEventListener('change', nextAppState => {
            if (appState.current.match(/inactive|background/) && nextAppState === 'active') {
                getPendingEventInvites()
                getFriendList()
                getPendingFriendInvites()
            }
            appState.current = nextAppState;
        });

        return () => {
            subscription.remove();
        };
    }, []);
    return (
        <AppContext.Provider value={{}}>
            {children}
        </AppContext.Provider>
    );
}