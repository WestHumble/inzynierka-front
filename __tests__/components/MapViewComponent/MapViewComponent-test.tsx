import React from 'react';
import renderer from 'react-test-renderer';
import { expect, test } from '@jest/globals';
import MapViewComponent from "../../../src/components/MapViewComponent/MapViewComponent";
import {EventsContext} from "../../../src/context/EventsContext";
import {LocationContext} from "../../../src/context/LocationContext";
import {FriendsContext} from "../../../src/context/FriendsContext";
import {ApiContext} from "../../../src/context/ApiContext";

const mockedNavigate = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
    useIsFocused: () => ({
      isFocused: true,
    }),
  };
});
jest.mock('react-native-maps', () => {
  const { View } = require('react-native');
  const React = require('react');

  const MockMapView = React.forwardRef(
      ({ testID, children, ...props }, ref) => {
        if (ref?.current) {
          ref.current = {
            getMapBoundaries: async () => ({
              northEast: {
                latitude: 2,
                longitude: 2,
              },
              southWest: {
                latitude: 1,
                longitude: 1,
              },
            }),
            getCamera: async () => ({
              center: {
                latitude: 2,
                longitude: 2,
              },
              heading: 1,
              pitch: 1,
              zoom: 1,
              altitude: 1,
            }),
            animateCamera: () => {},
          };
        }
        return (
            <View testID={testID} {...props}>
              {children}
            </View>
        );
      },
  );

  const MockMarker = React.forwardRef(({ testID, children, ...props }, ref) => {
    if (ref?.current) {
      ref.current = {
        redraw: () => {},
      };
    }

    return (
        <View testID={testID} {...props}>
          {children}
        </View>
    );
  });

  const mockMapTypes = {
    STANDARD: 0,
    SATELLITE: 1,
    HYBRID: 2,
    TERRAIN: 3,
    NONE: 4,
    MUTEDSTANDARD: 5,
  };

  return {
    __esModule: true,
    default: MockMapView,
    Marker: MockMarker,
    MAP_TYPES: mockMapTypes,
    PROVIDER_DEFAULT: 'default',
    PROVIDER_GOOGLE: 'google',
  };
});
test('renders correctly', () => {
  const eventsCreated = [];
  const eventsInvited = [];
  const eventsOther = [];
  const eventsCreatedSearch = [];
  const eventsInvitedSearch = [];
  const eventsOtherSearch = [];
  const isSearchActive = false;
  const userLocation = null;
  const friends = [];
  const userToken = null;

  const eventsRef = {current: []}
  const friendsRef = {current: []}
  const mapViewRef = {current: null}
  const get = async (endpoint, params? = null, success = null, error = null, noAuth = false, customConfig = null)=> {}
  const loadCloseEvents = ()=>{}
  const clearSearchEvents = ()=>{}

  const tree = renderer.create(
      <ApiContext.Provider value={{get, userToken}}>
        <LocationContext.Provider value={{userLocation,eventsRef,friendsRef,mapViewRef}}>
          <FriendsContext.Provider value={{friends}}>
            <EventsContext.Provider value={{eventsCreated,
              eventsInvited,
              eventsOther,
              eventsCreatedSearch,
              eventsInvitedSearch,
              eventsOtherSearch,
              loadCloseEvents,
              clearSearchEvents,
              isSearchActive}}>
              <MapViewComponent/>
            </EventsContext.Provider>
          </FriendsContext.Provider>
        </LocationContext.Provider>
      </ApiContext.Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});