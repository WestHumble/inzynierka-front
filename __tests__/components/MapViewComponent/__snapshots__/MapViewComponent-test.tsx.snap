// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <View
    customMapStyle={
      [
        {
          "featureType": "poi.attraction",
          "stylers": [
            {
              "visibility": "off",
            },
          ],
        },
        {
          "featureType": "poi.business",
          "stylers": [
            {
              "visibility": "off",
            },
          ],
        },
        {
          "elementType": "geometry.fill",
          "featureType": "poi.business",
          "stylers": [
            {
              "visibility": "off",
            },
          ],
        },
        {
          "elementType": "geometry.stroke",
          "featureType": "poi.business",
          "stylers": [
            {
              "visibility": "off",
            },
          ],
        },
        {
          "elementType": "labels.text",
          "featureType": "poi.business",
          "stylers": [
            {
              "visibility": "off",
            },
          ],
        },
        {
          "elementType": "labels.text.fill",
          "featureType": "poi.business",
          "stylers": [
            {
              "visibility": "off",
            },
          ],
        },
        {
          "elementType": "labels.text.stroke",
          "featureType": "poi.business",
          "stylers": [
            {
              "visibility": "off",
            },
          ],
        },
        {
          "featureType": "poi.government",
          "stylers": [
            {
              "visibility": "off",
            },
          ],
        },
        {
          "featureType": "poi.medical",
          "stylers": [
            {
              "visibility": "off",
            },
          ],
        },
        {
          "featureType": "poi.school",
          "stylers": [
            {
              "visibility": "off",
            },
          ],
        },
        {
          "featureType": "poi.sports_complex",
          "stylers": [
            {
              "visibility": "off",
            },
          ],
        },
      ]
    }
    followsUserLocation={true}
    initialRegion={
      {
        "latitude": 52.4064,
        "latitudeDelta": 0.14,
        "longitude": 16.9252,
        "longitudeDelta": 0.16,
      }
    }
    onRegionChange={[Function]}
    provider="google"
    rotateEnabled={false}
    showsUserLocation={true}
    style={
      {
        "elevation": 1,
        "height": "100%",
        "position": "absolute",
        "width": "100%",
        "zIndex": -1,
      }
    }
    toolbarEnabled={false}
    userInterfaceStyle="dark"
  />,
  <View
    style={
      {
        "alignContent": "space-between",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "height": 215,
        "justifyContent": "flex-end",
        "position": "absolute",
        "right": "5%",
        "top": "57%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#F2B138",
          "borderColor": "#131417",
          "borderRadius": 50,
          "borderStyle": "solid",
          "borderWidth": 3,
          "marginVertical": 5,
          "padding": 12,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Image
          source={1}
          style={
            {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#F2B138",
          "borderColor": "#131417",
          "borderRadius": 50,
          "borderStyle": "solid",
          "borderWidth": 3,
          "marginVertical": 5,
          "padding": 12,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Image
          source={1}
          style={
            {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
    </View>
  </View>,
]
`;
